---
title: "HT promoters"
author: "Claire Rioualen"
date: '`r Sys.Date()`'
output:
 html_document:
   fig_caption: yes
   highlight: zenburn
   self_contained: yes
   theme: cerulean
   toc: yes
   toc_depth: 5
   toc_float: yes
   number_sections: no
   code_folding: hide


---
<style type="text/css">

body{ /* Normal  */
      font-size: 14px;
  }
</style>

```{r init, include = FALSE, echo = FALSE}
setwd("/Users/rioualen/Google Drive/Work/Ecoli_mapping_project/Feature_set")
```


```{r setup}
knitr::opts_knit$set(root.dir = "~/Google Drive/Work/Ecoli_mapping_project/Feature_set")
knitr::opts_chunk$set(echo = TRUE, include = TRUE, warning = FALSE, message = FALSE, cache = FALSE, eval = TRUE, fig.width = 12, fig.height = 9, fig.align = 'center')

```

```{r libs-n-functions}
library(ggplot2)
library(dplyr)
library(DT)
library(gridExtra)
library(tidyr)
library(UpSetR)

devtools::load_all("~/Desktop/Git/EcoliGenes")

concat_uniq <- function(x, sep = ","){
  paste0(unique(na.omit(x)), collapse = sep)
}

check_dup <- function(df){
  test_dup <- df %>%
    mutate(coords = paste0(start, "_", stop, "_", strand)) %>%
    group_by(coords) %>%
    summarise(count = n(), 
              across(where(is.character), concat_uniq, sep = "_"),
              across(where(is.numeric), concat_uniq, sep = "_"),
              .groups = 'drop') %>%
    arrange(desc(count)) %>%
    filter(count>1) 
  test_dup
}
```

> OJO
Por ahora hice mis propias tablas de metadatos para generar las colecciones, y descargué los archivos fuentes en mi carpeta de trabajo. Más adelante, se debería considerar la forma de “fusionarlo” con las tablas de metadatos de curación, cuidando la congruencia de los datasets y condiciones, y la ubicación de los archivos fuentes. 


# Transcription start sites from high-throughput experiments

<!-- ## To do -->

<!-- * Fields to add or remove -->
<!--   * Sequence? -->
<!--   * Gene: keep genes reported by authors or drop them and map ourselves? -->
<!--   * Orientation? -->
<!--   * ... -->

<!-- * Missing metadata -->
<!--   * Conditions of the Conway dataset -->

<!-- * Make decision about duplicates (Morett, Storz, Yan) -->
<!--   * Depends on above decision about author-reported genes  -->

<!-- * Graph distances TSS-closest gene -->
<!--   * Make function to get closest gene start -->
<!--   * Some distributions look odd, should revise the code -->

## Format 

Here, a **dataset** is defined by a combination of **PMID and experimental condition**. 

TSSs are collected from a variety of sources and processed to produce uniform datasets with the following fields:

* **id:** Unique TSS ID composed as follows: `TSS_<author>_<condition>_<num>`
* **start:** Left genomic position. If not reported, will be the same as `pos_1`
* **stop:**  Right genomic position. If not reported, will be the same as `pos_1`
* **strand:** TSS strand reported by authors
* **pos_1:** TSS +1 position reported by authors
* **gene:** Gene(s) associated to the TSS, as reported by authors (*); can be empty
* **orientation:** as reported by authors (*); can be empty
* **total_reads:** as reported by authors (*); can be empty

## Metadata

```{r sources}
## Read datasets table
ht_tss_metadata <- read.delim("RegulonDB_HT/metadata/HT_TSS_datasets.tsv", comment.char = "#", header = T, na.strings = c("NA", "")) %>%
  dplyr::filter(!is.na(condition)) %>%
  dplyr::mutate(dataset_name = paste0(author, "_", make.names(condition)))

## Display table
DT <- DT::datatable(ht_tss_metadata, options = list(dom = '', pageLength = 20))
DT

## Make list from table
ht_tss_dataset_list <- setNames(split(ht_tss_metadata, seq(nrow(ht_tss_metadata))), ht_tss_metadata$dataset_id)
```

## Datasets per author {.tabset}

### Conway

Reference article: [Conway et al., 2014](https://doi.org/10.1128/mbio.01442-14)

Conditions:

* **Pending curation**

Notes:

* Files were downloaded on 2021/10/01
* Headers had to be altered manually to fit proper table delimitations, and genomic positions extracted from the "Promoter" column before connversion
* Genome coordinates were converted from version U00096.2 to version U00096.3 [here](https://biocyc.org/ECOLI/map-seq-coords-form?chromosome=COLI-K12) 

```{r ht-conway}
## Code executed just once, in order to export promoters coordinates and convert them through Ecocyc
# Conway_temp <- read.delim("HT_data_source/Conway/mbo004141900st2.tsv", comment.char = "#", header = T) %>%
#   dplyr::mutate(Promoter = gsub("^[PSIAS]{1,2}-", "", Promoter)) %>%
#   dplyr::select(Num, Promoter) %>%
#   write.table(file = paste0("HT_data_source/Conway/temp_promoters.tsv"), sep = "\t", quote = F, col.names = T, row.names = F)


Conway_unknown <- read.delim("HT_data_source/Conway/mbo004141900st2.tsv", comment.char = "#", header = T) %>%
  dplyr::left_join(read.delim("HT_data_source/Conway/promoters_converted.tsv", comment.char = "#", header = T), by = "Num") %>%
  dplyr::rename(strand = STRAND, pos_1 = Promoter.y, gene = Gene) %>%
  dplyr::mutate(start = pos_1, stop = pos_1, orientation = NA, total_reads = NA) 

```


### Ju

Reference article: [Ju et al., 2019](https://doi.org/10.1038/s41564-019-0500-z)

Conditions:

* exponential phase
* stationary phase

Notes:

* Files were downloaded on 2021/09/30

```{r ht-ju}
Ju_all <- read.delim("HT_data_source/Ju/41564_2019_500_MOESM3_ESM.tsv", comment.char = "#", header = T, na.strings = c("NA", "")) %>%
  dplyr::rename(strand = TSS_direction, pos_1 = TSS_site) %>%
  dplyr::mutate(gene = NA, start = pos_1, stop = pos_1, orientation = NA, total_reads = NA) %>%
  dplyr::distinct()


##---------------------------------------------------------------------------------
Ju_exponential <- Ju_all %>%
  dplyr::filter(!is.na(Detected_in_log_phase))

Ju_stationary <- Ju_all %>%
  dplyr::filter(!is.na(Detected_in_stationary_phase)) 
```

### Morett

Reference article: [Mendoza-Vargas et al., 2009](https://doi.org/10.1371/journal.pone.0007526)

Conditions:

* 5 tri or monophosphate enrichment
* 454 data
* RACE data

Notes:

* Files were downloaded on 2021/03/19 [here](http://regulondb.ccg.unam.mx/highthroughputdatasetssearch?term=all) 
* Genome coordinates were converted from version U00096.2 to version U00096.3 [here](https://biocyc.org/ECOLI/map-seq-coords-form?chromosome=COLI-K12) 
* 3 datasets files formatted differently:
  * One file has TSS left and right positions, as well as "max frequency" position.It also has "orientation" information. 
  * Two files have gene's left and right coordinates, and relative position of the TSS to the gene (unique position is calculated depending on strand). These files don't have relative orientation information
  
* **Some TSSs are duplicated for they're associated to different genes, their position and strand are the same. To be merged?**
  
```{r ht-morett}
## Dataset 225: promoter position = max_frequency column
##---------------------------------------------------------------------------------

Morett_phosphate.enrichment <- read.delim("HT_data_source/Morett/HTRI00000225_HighThroughputDataSets_converted.txt", comment.char = "#", header = F) %>% 
  dplyr::rename(gene = V10, start = V4, stop = V5, pos_1 = V6, strand = V9, orientation = V11, total_reads = V8) %>%
  dplyr::mutate(strand = ifelse(strand == "reverse", "-", ifelse(strand == "forward", "+", NA)))

## Datasets 229 and 230: promoter position = gene start + relative orientation
##---------------------------------------------------------------------------------

Morett_data.454 <- read.delim("HT_data_source/Morett/HTRI00000229_HighThroughputDataSets_converted.txt", comment.char = "#", header = F) %>% 
  dplyr::rename(gene = V10, strand = V9, left = V4, right = V5, relative_pos = V11, total_reads = V8) %>%
  dplyr::mutate(pos_1 = ifelse(strand == "forward", left + relative_pos, ifelse(strand == "reverse", right - relative_pos, NA)),
                start = pos_1, stop = pos_1, orientation = NA)  %>%
  dplyr::mutate(strand = ifelse(strand == "reverse", "-", ifelse(strand == "forward", "+", NA))) %>%
  dplyr::distinct()

Morett_data.RACE <- read.delim("HT_data_source/Morett/HTRI00000230_HighThroughputDataSets_converted.txt", comment.char = "#", header = F) %>% 
  dplyr::rename(gene = V10, strand = V9, left = V4, right = V5, relative_pos = V11, total_reads = V8) %>%
  dplyr::mutate(pos_1 = ifelse(strand == "forward", left + relative_pos, ifelse(strand == "reverse", right - relative_pos, NA)),
                start = pos_1, stop = pos_1, orientation = NA)  %>%
  dplyr::mutate(strand = ifelse(strand == "reverse", "-", ifelse(strand == "forward", "+", NA))) %>%
  dplyr::distinct()

```

### Palsson

Reference article: [Cho et al., 2014](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3923258/)

Conditions:

* Exponential phase (from [2012 paper](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3415461/)*)
* Glutamine as source of nitrogen
* Heat shock
* Stationary phase

Notes

* Genome coordinates were converted from version U00096.2 to version U00096.3 [here](https://biocyc.org/ECOLI/map-seq-coords-form?chromosome=COLI-K12) 


```{r pallson-tss}
##---------------------------------------------------------------------------------
Palsson_all <- read.delim("HT_data_source/Pallson/Palsson_2014_converted.tsv", sep = "\t", header = TRUE, stringsAsFactors = F) %>%
  tidyr::separate_rows(Conditions, sep = "") %>%
  dplyr::mutate_all(dplyr::na_if,"") %>%
  dplyr::filter(!is.na(Conditions)) %>%
  dplyr::rename(strand = Strand, pos_1 = TSS.position, total_reads = TSS.reads, condition = Conditions) %>%
  dplyr::mutate(gene = NA, start = pos_1, stop = pos_1, orientation = NA)

##---------------------------------------------------------------------------------
Palsson_exponential <- Palsson_all %>%
  dplyr::filter(condition == "E") 

Palsson_glutamine.nitrogen <- Palsson_all %>%
  dplyr::filter(condition == "G")

Palsson_heat.shock <- Palsson_all %>%
  dplyr::filter(condition == "H")

Palsson_stationary <- Palsson_all %>%
  dplyr::filter(condition == "S") 

```

### Storz 

Reference article: [Thomason et al., 2014](https://doi.org/10.1128/jb.02096-14)

Conditions:

* LB growth medium, OD = 2.0 (stationary phase)
* LB growth medium, OD = 0.4 (exponential phase)
* Minimum growth medium, OD = 0.4 (exponential phase)

Notes:

* Files were downloaded on 2021/03/19 [here](http://regulondb.ccg.unam.mx/highthroughputdatasetssearch?term=all)
* Genome coordinates were converted from version U00096.2 to version U00096.3 [here](https://biocyc.org/ECOLI/map-seq-coords-form?chromosome=COLI-K12) 
* **Some TSSs are duplicated for they're associated to different genes, their position and strand are the same. To be merged?**

```{r ht-storz}
##---------------------------------------------------------------------------------
Storz_rich.stationary <- read.delim("HT_data_source/Storz/HTRI00000227_HighThroughputDataSets_converted.txt", comment.char = "#", header = F) %>% 
  dplyr::rename(gene = V10, start = V4, strand = V9, orientation = V11, total_reads = V8) %>%
  dplyr::mutate(stop = start, pos_1 = start)  

Storz_mini.exponential <- read.delim("HT_data_source/Storz/HTRI00000228_HighThroughputDataSets_converted.txt", comment.char = "#", header = F) %>% 
  dplyr::rename(gene = V10, start = V4, strand = V9, orientation = V11, total_reads = V8) %>%
  dplyr::mutate(stop = start, pos_1 = start) 

Storz_rich.exponential <- read.delim("HT_data_source/Storz/HTRI00000458_HighThroughputDataSets_converted.txt", comment.char = "#", header = F) %>% 
  dplyr::rename(gene = V10, start = V4, strand = V9, orientation = V11, total_reads = V8) %>%
  dplyr::mutate(stop = start, pos_1 = start)

```

### Wade

Data not published, refer to [this article](https://doi.org/10.1007/978-1-62703-730-3_1) for details on methods

Conditions:

* M9 + 0.2% glycerol, cells grown with shaking at 30°C

Notes:

* TSSs queried from Galagan database on 2021-09-21

```{r wade-tss}
##---------------------------------------------------------------------------------
Wade_mini <- read.delim("HT_data_source/Wade/Wade_promoters_21-09-21.tsv", comment.char = "#", header = T) %>%
  dplyr::mutate(pos_1 = start, gene = NA, orientation = NA, total_reads = NA)

```

### Yan 

Reference paper: [Yan et al., 2018](10.1038/s41467-018-05997-6)

Conditions:

* M9 (minimal medium)
* Rich

Notes:

* HT-inferred TSSs from PacBio long read data.
* Files downloaded [there](https://drive.google.com/drive/folders/156UQL5wUEPbAIKeSDtVoVml0vG4m4j8T?usp=sharing)
* **Some TSSs are duplicated for they're associated to different TUs, their position 1 and strand are the same. To be merged?**


```{r pacbio-tss}
##---------------------------------------------------------------------------------
Yan_mini <- read.table("HT_data_source/Yan/M9_RegulonDB_TU_definedEnd", stringsAsFactors = F) %>% dplyr::mutate(type = "definedEnd") %>% 
  rbind.data.frame(read.table("HT_data_source/Yan/M9_RegulonDB_TU_longestRead", stringsAsFactors = F) %>% dplyr::mutate(type = "longestRead")) %>%
  dplyr::rename(gene = V8, start = V1, stop = V2, strand = V3) %>%
  dplyr::mutate(pos_1 = ifelse(strand == "-", stop, start), stop = pos_1, start = pos_1, orientation = NA, total_reads = NA)

##---------------------------------------------------------------------------------
Yan_rich <- read.table("HT_data_source/Yan/Rich_RegulonDB_TU_definedEnd", stringsAsFactors = F) %>% dplyr::mutate(type = "definedEnd") %>% 
  rbind.data.frame(read.table("HT_data_source/Yan/Rich_RegulonDB_TU_longestRead", stringsAsFactors = F) %>% dplyr::mutate(type = "longestRead")) %>%
  dplyr::rename(gene = V8, start = V1, stop = V2, strand = V3) %>%
  dplyr::mutate(pos_1 = ifelse(strand == "-", stop, start), stop = pos_1, start = pos_1, orientation = NA, total_reads = NA)

```


## Uniformization

* Generate unique IDs for TSSs
* Write one file per dataset

```{r write-files}
tss_dir <- "RegulonDB_HT/TSS"
dir.create(tss_dir, showWarnings = F)

## Uniformize format, add TSS ID column, write file
for(ds_id in names(ht_tss_dataset_list)) {
  ds_name <- ht_tss_dataset_list[[ds_id]]$dataset_name
  df <- get(ds_name) %>%
    dplyr::arrange(start) %>%
    dplyr::mutate(id = paste0("TSS_", ds_name, "_", dplyr::row_number()))

  write.table(df %>% dplyr::select(id, start, stop, strand, pos_1, gene, orientation, total_reads),
              file = paste0(tss_dir, "/", ds_id, ".tsv"),
              sep = "\t", quote = F, col.names = T, row.names = F)
  
  assign(ds_name, df)
  
  ## Should be made a check/warning only once decision on duplicates is made
  # df2 <- get(ds_name) %>%
  #   dplyr::arrange(start) %>%
  #   dplyr::mutate(id = paste0(ds_name, "_", dplyr::row_number()))
  # write.table(check_dup(df2), file = paste0("RegulonDB_HT/TSS/", ds_id, "_duplicated.tsv"), sep = "\t", quote = F, col.names = T, row.names = F)
  ##---
}

```


## Results

### Summary of datasets

```{r stats-tss}
ht_tss_stats <- ht_tss_metadata %>%
  dplyr::rowwise() %>%
  dplyr::mutate(tss_num = nrow(get(dataset_name))) %>%
  dplyr::select(dataset_id, dataset_name, author, condition, tss_num)

DT2 <- DT::datatable(ht_tss_stats, options = list(dom = '', pageLength = 20)) %>% DT::formatRound('tss_num', digits=0)
DT2

```

### Distribution

```{r fig-stats-tss-1}

ggplot(ht_tss_stats, aes(x = dataset_name, y = tss_num, fill = author, color = condition)) +
  geom_col(size = 2) +
  scale_fill_manual(values = RColorBrewer::brewer.pal(7, "Pastel2")) +
  theme(axis.text.x = element_text(angle = 45, hjust=1), text = element_text(size = 14)) +
  ylab("Number of TSSs") +
  xlab("Dataset name") +
  labs(title = "Number of TSSs per dataset", subtitle = "Coloured by authors and conditions")
  
```

### Distance to closest gene

* **Double-check that all datasets have coordinates up to the latest genome version**
* **Deal with positions that are equally distant to several genes**

```{r fig-stats-tss-3, fig.height = 36}
for(ds_id in names(ht_tss_dataset_list)) {
  ds_name <- ht_tss_dataset_list[[ds_id]]$dataset_name
  df <- get(ds_name) %>%
    dplyr::mutate(close = EcoliGenes::closest_gene(pos_1, strand)) %>%
    dplyr::mutate(distance = as.numeric(paste0(strand, 1)) * (pos_1 - close))

  assign(ds_name, df)
}

tss_distance_plots <- list()
for(ds_id in names(ht_tss_dataset_list)) {
  ds_name <- ht_tss_dataset_list[[ds_id]]$dataset_name
  df <- get(ds_name)
  
  gg <- ggplot(df, aes(distance)) + 
	  geom_histogram(binwidth = 10) + 
    geom_vline(aes(xintercept = 0, color = "red")) +
    geom_vline(aes(xintercept = -100, color = "red")) +
    theme(legend.position = "none") +
	  scale_x_continuous(limits = c(-1000,500)) +
    ylab("Number of TSSs") +
    xlab("Distance to closest gene start") +
    ggtitle(ds_name)
  tss_distance_plots[[ds_id]] <- gg
}
do.call("grid.arrange", c(tss_distance_plots, nrow = length(ht_tss_dataset_list)))
```

